Support for NSS in latest firefox

I have run firefox and tried to figure out the dynamically linked ssl libraries.
```
pegasus@pegasus:~$ sudo lsof -p 21129 | grep -iE 'ssl|nss'
[sudo] password for pegasus: 
lsof: WARNING: can't stat() fuse.portal file system /run/user/1000/doc
      Output information may be incomplete.
lsof: WARNING: can't stat() fuse.gvfsd-fuse file system /run/user/1000/gvfs
      Output information may be incomplete.
firefox-b 21129 pegasus  mem       REG              259,7    567848 32653234 /opt/firefox/libnssckbi.so
firefox-b 21129 pegasus  mem       REG              259,7    447160 32653231 /opt/firefox/libssl3.so
firefox-b 21129 pegasus  mem       REG              259,7    776816 32653277 /opt/firefox/libnss3.so
firefox-b 21129 pegasus  mem       REG              259,7    189080 32653243 /opt/firefox/libnssutil3.so
```

If I add this path in libresolver.c:
```
const char *COMMON_PATHS[] = {
    "/opt/firefox/",
    "/home/linuxbrew/.linuxbrew/",
    "/lib/",
    "/lib64/",
    "/lib32/",
    "/usr/lib/",
    "/usr/lib64/",
    "/usr/lib32/",
    "/usr/local/lib/",
    
};
```
and compile and run the tool and visit any big websites, it gives segmentation fault.
```
StreamTrans #66��M��h{
    "componentChunkName":�---src-pages-about-mdx",
    "path": "/�/(result": {"9TContext":{"frontmatterLtitle":"About","bann�\"/assets/bg/bg3.jpg"}}},	vXstaticQueryHashes": []}crb72c0lA2suMWPWT92PFLaRvFrFn9HVzI6Vh50YZgB3AObSMWNAd4zBEEEG13G5zsHSQPaWhIb7uocyHf0eN45QAAABjkN89pQAAAQDAEgwRgIhAPJQX4QArFCjM0sKKzsWLmqmmU8lMhKEYR2Tges1AQyQAiEA2Y3VhP5RG+dapcbwYgVbrTlgWzO7KE/lg1x11CVcz3QwDQYJKoZIhvcNAQELBQADggEBAHKlvzObJBxxgyLaUNCEFf37mNFsUtXmaWvkmcfIt9V+TZ7Qmtvjx5bsd5lqAflp/eqk4+JYpnYcKWrZfM/vMdxPQTeh/VQWewY/hYn6X/V1s�@B"�otVd<{���
Worker Launcher�sNaPpY�2IS7MIg7wKR+DDxoNj4fFFluxteVNgbtwuJcb23NkBQqfHXCvQWqxXZZA4Nwl/WoGPoGGdW5qVOc3BlhtITW53ASyhvKC7HArhj7LwQH8C/dRgn1agIHP9vVJ1NaZnPXhK98Tohv++OO0E/F/bVGNWVnLBQ4v5PjQzRQUTGvM2mUTAQAEAAAAAAABAQAAAAAAAAZ4MjU1MTkAAAAOUlNBLVBTUy1TSEEyNTYAA2YKMiaRXE/7uyCJhaYy3wW9w5eaVCJM1YWJaWtuluqDAAAAAAAABz0wggc5MIIGIaADAgECAhAGPUkXQE055RPLP+7NGy4bMA0GCSqGSIb3DQEBCwUAMFkxCzAJBgNVBAYTAlVTMRUwEwYDVQQKEwxEaWdpQ2VydCBJbmMxMzAxBgNVBAMTKkRpZ2lDZXJ0IEdsb2JhbCBHMiBUTFMgUlNBIFNIQTI1NiAyMDIwIENBMTAeFw0yNDAzMTUwMDAwMDBaFw0yNTAzMTQyMzU5NTlaMGcxCzAJBgNVB�@B�RQYDV�Ǜr�
Socket ThreadM1TYW4gRnJhbmNpc2NvMRUwEwYDVQQKEwxHaXRIdWIsIEluYy4xFDASBgNVBAMMCyouZ2l0aHViLmlvMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEArSsUpTpMQa+4sJjdk65eUb7eN6sPoQ/WBzWp7fmDrwWrIa5U85R11g1mLKaNgxnHLCg2nerGVsUUFN/162xrJq9P65b7ZQyOoKi0B0oqJwESym4TGgAIW42BOLuxJRPsDnn6Tj/7k75W2lrFDl2ZCTsfFyq8xjHmjAFT58HBgMP6Fd6Ddi/Etk14iU3w6WpYvzD0dsb7dxx6BUSM4lBuStytbshAyrZST3ZePEg+YxUi9p5+pwLWmgZi9LhW8SHfHri8krWEQzhgswoFoT+GoW1wyjOL4f/wmpMJ/M9CGe7bUciin2tK5zHGdlt70B4fPYsRGlRN/euOA4yD08HVFQIDAQABo4ID7TCCA+kwHwYDVR0jBBgwFoAUdIWAwGbH3zfez70pN6oDHb7tzRcwHQYDVR0OBBYEFOhvV+uGUZjrn6W+U9rblKwoLvvtMHsGA1UdEQR0MHKCCyouZ2l0aHViLmlvgglnaXRodWIuaW+CFWdpdGh1YnVzZXJjb250ZW50LmNvbYIOd3d3LmdpdGh1Yi5jb22CDCouZ2l0aHViLmNvbYIXKi5naXRodWJ1c2VyY29udGVudC5jb22CCmdpdGh1Yi5jb20wPgYDVR0gBDcwNTAzBgZngQwBAgIwKTAnBggrBgEFBQcCARYbaHR0cDovL3d3dy5kaWdpY2VydC5jb20vQ1BTMA4GA1UdDwEB/wQEAwIFoDAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwgZ8GA1UdHwSBlzCBlDBIoEagRIZCaHR0cDovL2NybDMuZGlnaWNlcnQuY29tL0RpZ2lDZXJ0R2xvYmFsRzJUTFNSU0FTSEEyNTYyMDIwQ0ExLTEuY3JsMEigRqBEhkJodHRwOi8vY3JsNC5kaWdpY2VydC5jb20vRGlnaUNlcnRHbG9iYWxHMlRMU1JTQVNIQTI1NjIwMjBDQTEtMS5jcmwwgYcGCCsGAQUFBwEBBHsweTAkBggrBgEFBQcwAYYYaHR0cDovL29jc3AuZGlnaWNlcnQuY29tMFEGCCsGAQUFBzAChkVodHRwOi8vY2FjZXJ0cy5kaWdpY2VydC5jb20vRGlnaUNlcnRHbG9iYWxHMlRMU1JTQVNIQTI1NjIwMjBDQTEtMS5jcnQwDAYDVR0TAQH/BAIwADCCAX8GCisGAQQB1nkCBAIEggFvBIIBawFpAHYATnWjJ1yaEMM4W2zU3z9S6x3w4I4bjWnAsfpksWKaOd8AAAGOQ3z2gAAABAMARzBFAiBT8znbtZzHQpDcgjuQK`��ssl_sniffer: we might lose some data (%d), need some recursive read
Segmentation fault
```
Like this.
How to solve this?

No I'm using normal Ubuntu on host machine. This is the log I get when I ran ssl_sniffer tool first and then tried to open the firefox browser.
```
pegasus@pegasus:~/Documents/ssl-ebpf-projects/evilBPF/src/ssl_sniffer$ sudo ./sniffer 
[sudo] password for pegasus: 
libssl.so probes attached to /home/linuxbrew/.linuxbrew/lib/libssl.so
libgnutls.so probes attached to /home/linuxbrew/.linuxbrew/lib/libgnutls.so
libnspr4.so probes attached to /opt/firefox/libnspr4.so
Press Ctrl+C to stop
[+] firefox-bin(7455), ts: 249618064132, op: SSL_OP_READ, len: 10 --> 
1716997470
[+] firefox-bin(7455), ts: 249618739928, op: SSL_OP_READ, len: 10 --> 
1716692468
[+] firefox-bin(7455), ts: 249674068446, op: SSL_OP_READ, len: 409 --> 
[Install4F96D1932A9F858E]
Default=q42nr58k.default-release
Locked=1

[Profile1]
Name=default
IsRelative=1
Path=vwihe3br.default
Default=1

[Profile0]
Name=default-release
IsRelative=1
Path=q42nr58k.default-release

[General]
StartWithLastProfile=1
Version=2

[Install6AFDA46A1A8AD48]
Default=1gqab9so.default-release-1
Locked=1

[Profile2]
Name=default-release-1
IsRelative=1
Path=1gqab9so.default-release-1


[+] firefox-bin(7455), ts: 249681743715, op: SSL_OP_READ, len: 155 --> 
[Compatibility]
LastVersion=126.0.1_20240526221752/20240526221752
LastOSABI=Linux_x86_64-gcc3
LastPlatformDir=/opt/firefox
LastAppDir=/opt/firefox/browser

[+] firefox-bin(7455), ts: 249691168121, op: SSL_OP_READ, len: 429 --> 
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
//
// This pref is in its own file for complex reasons. See the comment in
// browser/app/Makefile.in, bug 756325, and bug 1431342 for details. Do not add
// other prefs to this file.

pref("app.update.channel", "release");

[+] Socket Thread(7455), ts: 249695022238, op: SSL_OP_WRITE, len: 1 --> 
I
[+] Socket Thread(7455), ts: 249695062537, op: SSL_OP_READ, len: 1 --> 
I
[+] firefox-bin(7455), ts: 249710729324, op: SSL_OP_READ, len: 902633 --> 
// Mozilla User Preferences

// DO NOT EDIT THIS FILE.
//
// If you make changes to this file while the application is running,
// the changes will be overwritten when the application exits.
//
// To change a preference value, you can either:
// - modify it via the UI (e.g. via about:config in the browser); or
// - set it within a user.js file in your profile.

user_pref("accessibility.typeaheadfind.flashBar", 0);
user_pref("app.normandy.first_run", false);
user_pref("app.normandy.migrationsApplied", 12);
user_pref("app.normandy.user_id", "4951b1e0-af67-479b-b226-002ca718e91d");
user_pref("app.update.download.attempts", 0);
user_pref("app.update.elevate.attempts", 0);
user_pref("app.update.lastUpdateTime.addon-background-update-timer", 1717366776);
user_pref("app.update.lastUpdateTime.background-update-timer", 1717447293);
user_pref("app.update.lastUpdateTime.browser-cleanup-thumbnails", 1717450893);
user_pref("app.update.lastUpdateTime.recipe-client-addon-run", 1717447293);
user_pref("app.update.lastUpdateTime.region-update-timer", 1716997195);
user_pref("app.update.lastUpdateTime.rs-experiment-loader-timer", 1717447293);
user_pref("app.update.lastUpdateTime.services-settings-poll-changes", 1717366776);
user_pref("app.update.lastUpdateTime.telemetry_modules_ping", 1717007892);
user_pref("app.update.lastUpdateTime.xpi-signature-verification", 1717366776);
user_pref("browser.bookmarks.addedImportButton", true);
user_pref("browser.bookmarks.defaultLocation", "iafKK8xHzjH9");
user_pref("browser.bookmarks.editDialog.confirmationHintShowCount", 3);
user_pref("browser.bookmarks.restore_default_bookmarks", false);
user_pref("browser.contentblocking.category", "standard");
user_pref("browser.contentblocking.cfr-milestone.milestone-achieved", 10000);
user_pref("browser.contentblocking.cfr-milestone.milestone-shown-time", "1716063623738");
user_pref("browser.contextual-services.contextId", "{9ff2e8ce-b03a-4332-8570-46d918e31cf2}");
user_pref("browser.download.lastDir", "/home/pegasus/Downloads");
user_pref("browser.download.panel.shown", true);
user_pref("browser.download.viewableInternally.typeWasRegistered.avif", true);
user_pref("browser.download.viewableInternally.typeWasRegistered.webp", true);
user_pref("browser.engagement.ctrlTab.has-used", true);
user_pref("browser.engagement.downloads-button.has-used", true);
user_pref("browser.engagement.fxa-toolbar-menu-button.has-used", true);
user_pref("browser.laterrun.bookkeeping.profileCreationTime", 1703182035);
user_pref("browser.laterrun.bookkeeping.sessionCount", 1);
user_pref("browser.laterrun.bookkeeping.updateAppliedTime", 1716997470);
user_pref("browser.migration.version", 144);
user_pref("browser.newtabpage.activity-stream.impressionId", "{9f183863-0592-4f3e-a3ad-f90f1f4c17f8}");
user_pref("browser.newtabpage.blocked", "{\"Z74nFw1gFDTLEkP9t+g2yA==\":1,\"icoPFNp05vcQbDNLOw7oVQ==\":1,\"q57v+BoW4sAxVabNiv28Hg==\":1,\"/tnIVhsVbVI1/+WRVvHhUw==\":1,\"4TjydvjqCFiF+dEsmLlk5g==\":1,\"U+9FiAGhYbL4cDg5HHviyw==\":1,\"91p9AS5sI68zIMjdpfaCYg==\":1,\"x7138edceycD4uSUY8fByg==\":1,\"WOMwJuSU2Uz9o197kOpyxQ==\":1,\"OL6iulNjOA696Wgt8vgWKQ==\":1}");
user_pref("browser.newtabpage.storageVersion", 1);
user_pref("browser.pageActions.persistedActions", "{\"ids\":[\"bookmark\"],\"idsInUrlbar\":[\"bookmark\"],\"idsInUrlbarPreProton\":[],\"version\":1}");
user_pref("browser.pagethumbnails.storage_version", 3);
user_pref("browser.proton.toolbar.version", 3);
user_pref("browser.region.update.updated", 1715507478);
user_pref("browser.rights.3.shown", true);
user_pref("browser.safebrowsing.provider.google4.lastupdatetime", "1717452838242");
user_pref("browser.safebrowsing.provider.google4.nextupdatetime", "1717454638242");
user_pref("browser.safebrowsing.provider.mozilla.lastupdatetime", "1717450854857");
user_pref("browser.safebrowsing.provider.mozilla.nextupdatetime", "1717472454857");
user_pref("browser.search.region", "BD");
user_pref("browser.sessionstore.upgradeBackup.latestBuildID", "20240526221752");
user_pref("browser.shell.didSkipDefaultBrowserCheckOnFirstRun", true);
user_pref("browser.shell.mostRecentDateSetAsDefault", "1717388323");
user_pref("browser.startup.couldRestoreSession.count", 2);
user_pref("browser.startup.homepage_override.buildID", "20240526221752");
user_pref("browser.startup.homepage_override.mstone", "126.0.1");
user_pref("browser.startup.lastColdStartupCheck", 1717388323);
user_pref("browser.topsites.contile.cacheValidFor", 10115);
user_pref("browser.topsites.contile.cachedTiles", "[]");
user_pref("browser.topsites.contile.lastFetch", 1717394943);
user_pref("browser.translations.panelShown", true);
user_pref("browser.uiCustomization.state", "{\"placements\":{\"widget-overflow-fixed-list\":[],\"unified-extensions-area\":[\"foxyproxy_eric_h_jung-browser-action\",\"wappalyzer_crunchlabz_com-browser-action\"],\"nav-bar\":[\"back-button\",\"forward-button\",\"stop-reload-button\",\"customizableui-special-spring1\",\"urlbar-container\",\"customizableui-special-spring2\",\"save-to-pocket-button\",\"downloads-button\",\"fxa-toolbar-menu-button\",\"unified-extensions-button\"],\"toolbar-menubar\":[\"menubar-items\"],\"TabsToolbar\":[\"firefox-view-button\",\"tabbrowser-tabs\",\"new-tab-button\",\"alltabs-button\"],\"PersonalToolbar\":[\"import-button\",\"personal-bookmarks\"]},\"seen\":[\"save-to-pocket-button\",\"developer-button\",\"foxyproxy_eric_h_jung-browser-action\",\"wappalyzer_crunchlabz_com-browser-action\"],\"dirtyAreaCache\":[\"nav-bar\",\"PersonalToolbar\",\"unified-extensions-area\",\"toolbar-menubar\",\"TabsToolbar\"],\"currentVersion\":20,\"newElementCount\":2}");
user_pref("browser.urlbar.placeholderName", "Google");
user_pref("browser.urlbar.placeholderName.private", "Google");
user_pref("browser.urlbar.quicksuggest.migrationVersion", 2);
user_pref("browser.urlbar.quicksuggest.scenario", "history");
user_pref("browser.urlbar.tipShownCount.searchTip_onboard", 4);
user_pref("browser.urlbar.tipShownCount.searchTip_redirect", 2);
user_pref("datareporting.policy.dataSubmissionPolicyAcceptedVersion", 2);
user_pref("datareporting.policy.dataSubmissionPolicyNotifiedTime", "1703182036843");
user_pref("devtools.debugger.expressions", "[{\"input\":\"!function(d){\\\"object\\\"==typeof exports&&\\\"undefined\\\"!=typeof module?module.exports=d():\\\"function\\\"==typeof define&&define.amd?define([],d):(\\\"undefined\\\"!=typeof window?window:\\\"undefined\\\"!=typeof global?global:\\\"undefined\\\"!=typeof self?self:this).TTAM=d()}(function(){var define,module,exports;return function n(i,o,a){function r(e,d){if(!o[e]){if(!i[e]){var t=\\\"function\\\"==typeof require&&require;if(!d&&t)return t(e,!0);if(A)return A(e,!0);throw(t=new Error(\\\"Cannot find module '\\\"+e+\\\"'\\\")).code=\\\"MODULE_NOT_FOUND\\\",t}t=o[e]={exports:{}},i[e][0].call(t.exports,function(d){return r(i[e][1][d]||d)},t,t.exports,n,i,o,a)}return o[e].exports}for(var A=\\\"function\\\"==typeof require&&require,d=0;d<a.length;d++)r(a[d]);return r}({1:[function(d,e,t){\\\"use strict\\\";e.exports={trackPageview:function(d,e){window.dataLayer&&window.dataLayer.push({event:\\\"gaEvent\\\",gaEventCategory:\\\"airpr\\\",gaEventAction:d+\\\"_pageview\\\",gaEventLabel:e,gaEventValue:0,gaEventNonInteractionBool:!0})}}},{}],2:[function(d,e,t){\\\"use strict\\\";var n=$(\\\".button-continue\\\"),i=$(\\\".auto-refund-input\\\"),o=$(\\\".auto-refund-form\\\");function a(){var t=!0;return i.each(function(){var d,e=$(this);\\\"\\\"===e.val()?(t&&(d=e)&&$(\\\"html,body\\\").animate({scrollTop:d.offset().top-36},333),t=!1,(d=e).addClass(\\\"error\\\"),d.prev(\\\"label\\\").addClass(\\\"error\\\")):((e=e).removeClass(\\\"error\\\"),e.prev(\\\"label\\\").removeClass(\\\"error\\\"))}),t}e.exports=function(){n.click(function(d){d.preventDefault(),a()&&o.submit()})}},{}],3:[function(d,e,t){\\\"use strict\\\";t.initDrawers=function(){var o=\\\"drawer-open\\\",a=\\\"drawer-transition\\\";$(document).on(\\\"click\\\",\\\"[data-drawer]\\\",function(){var d,e,t,n,i=$(this).attr(\\\"aria-controls\\\");return i=(e=(d=i).trim().split(/\\\\s+/)).map(function(d){return\\\"#\\\"+d}).join(\\\", \\\"),e=e.map(function(d){return'[data-drawer-indicator=\\\"'+d+'\\\"]'}).join(\\\", \\\"),i=$(i),t=$('[aria-controls=\\\"'+d+'\\\"]'),n=$(e).add(t).add(i),e=\\\"true\\\"===$(t[0]).attr(\\\"aria-expanded\\\"),n.addClass(a),e?i.slideUp(500,function(){t.attr(\\\"aria-expanded\\\",\\\"false\\\"),n.removeClass(a),n.removeClass(o)}):(i.slideDown(500,function(){t.attr(\\\"aria-expanded\\\",\\\"true\\\")}),n.addClass(o)),!1})},t.fdaModalSetUp=function(){var d=\\\"fda-main\\\",e=$(window),t=$(\\\".fda-info\\\"),n=$(\\\".fda-tabs\\\"),i=$(\\\".checkout-header\\\");0<d.length?e.scroll(_.debounce(function(){0<$(this).scrollTop()?(n.addClass(\\\"modal-scrolled\\\"),i.addClass(\\\"desktop-scrolled\\\")):(n.removeClass(\\\"modal-scrolled\\\"),i.removeClass(\\\"desktop-scrolled\\\"))},10)):t.scroll(_.debounce(function(){0<$(this).scrollTop()?(t.addClass(\\\"modal-scrolled-content\\\"),n.addClass(\\\"modal-scrolled\\\")):(t.removeClass(\\\"modal-scrolled-content\\\"),n.removeClass(\\\"modal-scrolled\\\"))},10));var o=$(\\\".fda-main\\\");o.removeClass(\\\"hide\\\"),$(\\\".fda-main-options-option\\\").click(function(d){d.preventDefault();var e,d=$(this).data(\\\"fda-option\\\");(e=$(\\\".fda-tabs [aria-controls=\\\"+d+\\\"]\\\")).trigger(\\\"click\\\"),o.fadeOut(333,function(){t.fadeIn(333),e.focus()})}),$(\\\".fda-tabs-tab\\\").click(function(){$(\\\".carrier-status-fda-container\\\").animate({scrollTop:0},250,\\\"swing\\\")})},t.initTabs=function(){var n=\\\"is-current-tab\\\",i=\\\"current-content\\\",o=37,a=38,r=39,A=40;function N(d){return d.closest('[role=\\\"tablist\\\"]')}function t(d){d=N(d);return $('[role=\\\"tab\\\"]',d)}function d(d){d.preventDefault();var d=$(this),e=d.closest(\\\".tabs-group\\\").attr(\\\"tab-group-tabs\\\");$(\\\"[tab-group-tab=\\\"+e+\\\"]\\\").not(d).removeClass(n).attr({\\\"aria-selected\\\":\\\"false\\\",tabindex:\\\"-1\\\"}),d.addClass(n),d.attr({\\\"aria-selected\\\":\\\"true\\\",tabindex:\\\"0\\\"}),d.focus();var d=d[0].hasAttribute(\\\"tabs-dropdown\\\")?d.val():d.attr(\\\"aria-controls\\\"),t=\\\"[tab-content=\\\"+e+\\\"-\\\"+d+\\\"]\\\";$(\\\"[tab-group-content=\\\"+e+\\\"]\\\").removeClass(i),setTimeout(function(){$(\\\"[tab-group-content=\\\"+e+\\\"]\\\").addClass(\\\"hide\\\"),$(t).removeClass(\\\"hide\\\"),setTimeout(function(){$(t).addClass(i)},33)},300)}function s(d){var e=t(d),d=e.index(d);(d+1<e.length?e.eq(d+1):e.eq(0)).focus()}function l(d){var e=t(d),d=e.index(d);(0<=d-1?e.eq(d-1):e.eq(e.length-1)).focus()}$(\\\"body\\\").unbind(\\\"click\\\").on(\\\"click\\\",\\\"[tabs-tab]\\\",d),$(\\\"body\\\").on(\\\"change\\\",\\\"[tabs-dropdown]\\\",d),$(\\\"body\\\").on(\\\"keydown\\\",\\\"[tabs-tab]\\\",function(d){var e=$(this),t=d.which;\\\"vertical\\\"===N(e).attr(\\\"aria-orientation\\\")?t===A?(d.preventDefault(),s(e)):t===a&&(d.preventDefault(),l(e)):t===r?s(e):t===o&&l(e)}),$(\\\"[tab-group-tabs]\\\").each(function(){var d,e;$(this)[0].hasAttribute(\\\"tab-default-id\\\")&&(d=$(this).attr(\\\"tab-default-id\\\"),e=$(this).attr(\\\"tab-group-tabs\\\"),$(\\\"[tab-group-tabs=\\\"+e+\\\"] [aria-controls=\\\"+d+\\\"]\\\").trigger(\\\"click\\\"))}),$(\\\"#brca, #arsacs\\\").addClass(\\\"current-content\\\").removeClass(\\\"hide\\\")}},{}],4:[function(d,e,t){\\\"use strict\\\";var i=d(\\\"./shlex\\\").parse_discount_info;const r=$(\\\".checkout-panel\\\");let A=null,o=null;var a=$(\\\".submit\\\"),N=$(\\\".js-expand-checkout-button\\\"),s=$(\\\".js-paypal-button\\\"),l=null;function c(){var d=TTAM.LANGUAGE_CODE.split(\\\"-\\\");return d[1]=d[1].toUpperCase(),d.join(\\\"-\\\")}function u(){var d=$(\\\".cartv2-items\\\");return $(\\\".js-add-product\\\").off(),$(\\\".js-remove-product\\\").off(),$(\\\".js-upgrade-product\\\").off(),d.empty(),d}function n(d){$(\\\".js-show-overlay\\\").off(),$(\\\".js-hide-overlay\\\").off(),$(\\\".learn-more-overlay\\\").off(),$(\\\".learn-more-overlay-wrapper\\\").off();var n=u();0<d.kits.length?($.each(d.kits,function(d,e){e.discount_info=i(e.discount_label);var t=$(\\\"#cartv2-\\\"+e.alias+\\\"-template\\\").html(),e=$(_.template(t,e));n.append(e)}),$(\\\"#cart-section-panel-left-container\\\").removeClass(\\\"empty\\\"),d.totals.items_discount?($(\\\".js-totals-subtotal\\\").removeClass(\\\"hide-discount-info\\\"),$(\\\".js-totals-savings\\\").removeClass(\\\"hide-discount-info\\\")):($(\\\".js-totals-subtotal\\\").addClass(\\\"hide-discount-info\\\"),$(\\\".js-totals-savings\\\").addClass(\\\"hide-discount-info\\\")),l()):($(\\\"#cart-section-panel-left-container\\\").addClass(\\\"empty\\\"),$(\\\".js-totals-subtotal\\\").addClass(\\\"hide-discount-info\\\"),$(\\\".js-totals-savings\\\").addClass(\\\"hide-discount-info\\\")),$(\\\".js-totals-subtotal-numkits\\\").text(\\\"Subtotal (\\\"+d.kits.length+\\\")\\\"),$(\\\".js-totals-subtotal-amount\\\").text(d.totals.items_total).formatCurrency({region:c()}),$(\\\".js-totals-savings-amount\\\").text(d.totals.items_discount).formatCurrency({region:c()}),$(\\\".js-totals-total-numkits\\\").text(\\\"Total (\\\"+d.kits.length+\\\")\\\"),$(\\\".js-totals-total-amount\\\").text(d.totals.items_total_with_discount).formatCurrency({region:c()}),a.prop(\\\"disabled\\\",0==d.kits.length),N.prop(\\\"disabled\\\",0==d.kits.length),s.prop(\\\"disabled\\\",0==d.kits.length),p(o),g(),o.emit(\\\"loading:stop\\\")}function p(n){$(\\\".js-add-product\\\").on(\\\"click\\\",function(d){n.emit(\\\"loading\\\");const e=$(d.currentTarget);var t=e.data(\\\"alias\\\").trim(),d=e.data(\\\"kit-url\\\");return $.ajax({type:\\\"POST\\\",url:d,data:{alias:t},dataType:\\\"json\\\"}).done(function(d){n.emit(\\\"cartv2:kit:operation\\\",d)}),!1}),$(\\\".js-remove-product\\\").on(\\\"click\\\",function(d){n.emit(\\\"loading\\\");const e=$(d.currentTarget);d=e.data(\\\"kit-url\\\");return $.ajax({type:\\\"DELETE\\\",url:d,dataType:\\\"json\\\"}).done(function(d){n.emit(\\\"cartv2:kit:operation\\\",d)}),!1}),$(\\\".js-upgrade-product\\\").on(\\\"click\\\",function(d){n.emit(\\\"loading\\\");const e=$(d.currentTarget);d=e.data(\\\"kit-url\\\");return $.ajax({type:\\\"POST\\\",url:d,dataType:\\\"json\\\"}).done(function(d){n.emit(\\\"cartv2:kit:operation\\\",d)}),!1})}const f=function(){$(\\\".expanded-checkout-wrapper .horizontal-button, #checkout-panel-mobile-veil\\\").off(\\\"click\\\"),$(\\\".expand-checkout-wrapper\\\").removeClass(\\\"expanded\\\"),$(\\\".expanded-checkout-wrapper\\\").removeClass(\\\"expanded\\\"),$(\\\"#checkout-panel-mobile-veil\\\").hide()};function h(){const e=r.offset().top;let t=$(window).width()<768;function d(){var d=t;t=$(window).width()<768,$(window).width()<768?(t&&!d&&$(window).unbind(\\\"scroll\\\"),r.addClass(\\\"sticky-enabled\\\").removeClass(\\\"sticky-disabled\\\").css(\\\"width\\\",\\\"100%\\\")):(f(),r.addClass(\\\"sticky-reset-for-calculation\\\"),r.css(\\\"width\\\",\\\"unset\\\"),d=r.outerWidth(),r.removeClass(\\\"sticky-reset-for-calculation\\\"),function(d){const t=function(){r.addClass(\\\"sticky-disabled\\\").removeClass(\\\"sticky-enabled\\\")},n=function(){r.addClass(\\\"sticky-enabled\\\").removeClass(\\\"sticky-disabled\\\")};n(),r.css(\\\"width\\\",d);const i=$(\\\"#cartv2-body\\\"),o=i.offset().top;var a=$(window).scrollTop();A=function(){var d=o+i.outerHeight(),e=r.offset().top+r.outerHeight();d<=e&&\\\"fixed\\\"===r.css(\\\"position\\\")?(a=$(window).scrollTop()-(e-d),t()):\\\"absolute\\\"===r.css(\\\"position\\\")&&$(window).scrollTop()<a&&n()},$(window).bind(\\\"scroll\\\",A)}((e,d)))}d(),$(window).resize(d)}function g(){function e(){return d.removeClass(\\\"is-open\\\"),!1}const d=$(\\\".learn-more-overlay\\\");$(\\\".js-show-overlay\\\").click(function(d){const e=$(d.currentTarget),t=$(\\\"#\\\"+e.data(\\\"overlay-id\\\"));return t&&t.addClass(\\\"is-open\\\"),!1}),$(\\\".js-hide-overlay\\\").click(e),$(document).on(\\\"keydown\\\",function(d){\\\"Escape\\\"==d.key&&e()}),$(\\\".learn-more-overlay\\\").click(function(){e()}),$(\\\".learn-more-overlay-wrapper\\\").click(function(d){d.stopPropagation()})}t.initialize=function(d,e){TTAM.cartv2&&(o=d,g(),p(d),d.on(\\\"cartv2:kit:operation\\\",n),$(\\\".product-cards.slick-carousel\\\").slick({arrows:!1,infinite:!1,centerMode:!0,variableWidth:!0,initialSlide:1,mobileFirst:!0,edgeFriction:0}),h(),$(\\\".js-expand-checkout-button, .expand-checkout-wrapper .horizontal-button\\\").on(\\\"click\\\",function(){$(\\\".expand-checkout-wrapper\\\").addClass(\\\"expanded\\\"),$(\\�@ce:firefox-bin
1716997470�@زo:firefox-bin
1716692468�@
            ��!:firefox-bin�[Install4F96D1932A9F858E]
Default=q42nr58k.default-release
Locked=1

[Profile1]
Name=default
IsRelative=1
Path=vwihe3br.default
Default=1

[Profile0]
Name=default-release
IsRelative=1
Path=q42nr58k.default-release

[General]
StartWithLastProfile=1
Version=2

[Install6AFDA46A1A8AD48]
Default=1gqab9so.default-release-1
Locked=1

[Profile2]
Name=default-release-1
IsRelative=1
Path=1gqab9so.default-release-1

�@c1":firefox-bin�[Compatibility]
LastVersion=126.0.1_20240526221752/20240526221752
LastOSABI=Linux_x86_64-gcc3
LastPlatformDir=/opt/firefox
LastAppDir=/opt/firefox/browser
�@y��":firefox-bin�/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
//
// This pref is in its own file for complex reasons. See the comment in
// browser/app/Makefile.in, bug 756325, and bug 1431342 for details. Do not add
// other prefs to this file.

pref("app.update.channel", "release");
// Mozilla User PreferencesSocket ThreadI�@lX�#:firefox-bin��

// DO NOT EDIT THIS FILE.
//
// If you make changes to this file while the application is running,
// the changes will be overwritten when the application exits.
//
// To change a preference value, you can either:
// - modify it via the UI (e.g. via about:config in the browser); or
// - set it within a user.js file in your profile.

user_pref("accessibility.typeaheadfind.flashBar", 0);
user_pref("app.normandy.first_run", false);
user_pref("app.normandy.migrationsApplied", 12);
user_pref("app.normandy.user_id", "4951b1e0-af67-479b-b226-002ca718e91d");
user_pref("app.update.download.attempts", 0);
user_pref("app.update.elevate.attempts", 0);
user_pref("app.update.lastUpdateTime.addon-background-update-timer", 1717366776);
user_pref("app.update.lastUpdateTime.background-update-timer", 1717447293);
user_pref("app.update.lastUpdateTime.browser-cleanup-thumbnails", 1717450893);
user_pref("app.update.lastUpdateTime.recipe-client-addon-run", 1717447293);
user_pref("app.update.lastUpdateTime.region-update-timer", 1716997195);
user_pref("app.update.lastUpdateTime.rs-experiment-loader-timer", 1717447293);
user_pref("app.update.lastUpdateTime.services-settings-poll-changes", 1717366776);
user_pref("app.update.lastUpdateTime.telemetry_modules_ping", 1717007892);
user_pref("app.update.lastUpdateTime.xpi-signature-verification", 1717366776);
user_pref("browser.bookmarks.addedImportButton", true);
user_pref("browser.bookmarks.defaultLocation", "iafKK8xHzjH9");
user_pref("browser.bookmarks.editDialog.confirmationHintShowCount", 3);
user_pref("browser.bookmarks.restore_default_bookmarks", false);
user_pref("browser.contentblocking.category", "standard");
user_pref("browser.contentblocking.cfr-milestone.milestone-achieved", 10000);
user_pref("browser.contentblocking.cfr-milestone.milestone-shown-time", "1716063623738");
user_pref("browser.contextual-services.contextId", "{9ff2e8ce-b03a-4332-8570-46d918e31cf2}");
user_pref("browser.download.lastDir", "/home/pegasus/Downloads");
user_pref("browser.download.panel.shown", true);
user_pref("browser.download.viewableInternally.typeWasRegistered.avif", true);
user_pref("browser.download.viewableInternally.typeWasRegistered.webp", true);
user_pref("browser.engagement.ctrlTab.has-used", true);
user_pref("browser.engagement.downloads-button.has-used", true);
user_pref("browser.engagement.fxa-toolbar-menu-button.has-used", true);
user_pref("browser.laterrun.bookkeeping.profileCreationTime", 1703182035);
user_pref("browser.laterrun.bookkeeping.sessionCount", 1);
user_pref("browser.laterrun.bookkeeping.updateAppliedTime", 1716997470);
user_pref("browser.migration.version", 144);
user_pref("browser.newtabpage.activity-stream.impressionId", "{9f183863-0592-4f3e-a3ad-f90f1f4c17f8}");
user_pref("browser.newtabpage.blocked", "{\"Z74nFw1gFDTLEkP9t+g2yA==\":1,\"icoPFNp05vcQbDNLOw7oVQ==\":1,\"q57v+BoW4sAxVabNiv28Hg==\":1,\"/tnIVhsVbVI1/+WRVvHhUw==\":1,\"4TjydvjqCFiF+dEsmLlk5g==\":1,\"U+9FiAGhYbL4cDg5HHviyw==\":1,\"91p9AS5sI68zIMjdpfaCYg==\":1,\"x7138edceycD4uSUY8fByg==\":1,\"WOMwJuSU2Uz9o197kOpyxQ==\":1,\"OL6iulNjOA696Wgt8vgWKQ==\":1}");
user_pref("browser.newtabpage.storageVersion", 1);
user_pref("browser.pageActions.persistedActions", "{\"ids\":[\"bookmark\"],\"idsInUrlbar\":[\"bookmark\"],\"idsInUrlbarPreProton\":[],\"version\":1}");
user_pref("browser.pagethumbnails.storage_version", 3);
user_pref("browser.proton.toolbar.version", 3);
user_pref("browser.region.update.updated", 1715507478);
user_pref("browser.rights.3.shown", true);
user_pref("browser.safebrowsing.provider.google4.lastupdatetime", "1717452838242");
user_pref("browser.safebrowsing.provider.google4.nextupdatetime", "1717454638242");
user_pref("browser.safebrowsing.provider.mozilla.lastupdatetime", "1717450854857");
user_pref("browser.safebrowsing.provider.mozilla.nextupdatetime", "1717472454857");
user_pref("browser.search.region", "BD");
user_pref("browser.sessionstore.upgradeBackup.latestBuildID", "20240526221752");
user_pref("browser.shell.didSkipDefaultBrowserCheckOnFirstRun", true);
user_pref("browser.shell.mostRecentDateSetAsDefault", "1717388323");
user_pref("browser.startup.couldRestoreSession.count", 2);
user_pref("browser.startup.homepage_override.buildID", "20240526221752");
user_pref("browser.startup.homepage_override.mstone", "126.0.1");
user_pref("browser.startup.lastColdStartupCheck", 1717388323);
user_pref("browser.topsites.contile.cacheValidFor", 10115);
user_pref("browser.topsites.contile.cachedTiles", "[]");
user_pref("browser.topsites.contile.lastFetch", 1717394943);
user_pref("browser.translations.panelShown", true);
user_pref("browser.uiCustomization.state", "{\"placements\":{\"widget-overflow-fixed-list\":[],\"unified-extensions-area\":[\"foxyproxy_eric_h_jung-browser-action\",\"wappalyzer_crunchlabz_com-browser-action\"],\"nav-bar\":[\"back-button\",\"forward-button\",\"stop-reload-button\",\"customizableui-special-spring1\",\"urlbar-container\",\"customizableui-special-spring2\",\"save-to-pocket-button\",\"downloads-button\",\"fxa-toolbar-menu-button\",\"unified-extensions-button\"],\"toolbar-menubar\":[\"menubar-items\"],\"TabsToolbar\":[\"firefox-view-button\",\"tabbrowser-tabs\",\"new-tab-button\",\"alltabs-button\"],\"PersonalToolbar\":[\"import-button\",\"personal-bookmarks\"]},\"seen\":[\"save-to-pocket-button\",\"developer-button\",\"foxyproxy_eric_h_jung-browser-action\",\"wappalyzer_crunchlabz_com-browser-action\"],\"dirtyAreaCache\":[\"nav-bar\",\"PersonalToolbar\",\"unified-extensions-area\",\"toolbar-menubar\",\"TabsToolbar\"],\"currentVersion\":20,\"newElementCount\":2}");
user_pref("browser.urlbar.placeholderName", "Google");
user_pref("browser.urlbar.placeholderName.private", "Google");
user_pref("browser.urlbar.quicksuggest.migrationVersion", 2);
user_pref("browser.urlbar.quicksuggest.scenario", "history");
user_pref("browser.urlbar.tipShownCount.searchTip_onboard", 4);
user_pref("browser.urlbar.tipShownCount.searchTip_redirect", 2);
user_pref("datareporting.policy.dataSubmissionPolicyAcceptedVersion", 2);
user_pref("datareporting.policy.dataSubmissionPolicyNotifiedTime", "1703182036843");
user_pref("devtools.debugger.expressions", "[{\"input\":\"!function(d){\\\"object\\\"==typeof exports&&\\\"undefined\\\"!=typeof module?module.exports=d():\\\"function\\\"==typeof define&&define.amd?define([],d):(\\\"undefined\\\"!=typeof window?window:\\\"undefined\\\"!=typeof global?global:\\\"undefined\\\"!=typeof self?self:this).TTAM=d()}(function(){var define,module,exports;return function n(i,o,a){function r(e,d){if(!o[e]){if(!i[e]){var t=\\\"function\\\"==typeof require&&require;if(!d&&t)return t(e,!0);if(A)return A(e,!0);throw(t=new Error(\\\"Cannot find module '\\\"+e+\\\"'\\\")).code=\\\"MODULE_NOT_FOUND\\\",t}t=o[e]={exports:{}},i[e][0].call(t.exports,function(d){return r(i[e][1][d]||d)},t,t.exports,n,i,o,a)}return o[e].exports}for(var A=\\\"function\\\"==typeof require&&require,d=0;d<a.length;d++)r(a[d]);return r}({1:[function(d,e,t){\\\"use strict\\\";e.exports={trackPageview:function(d,e){window.dataLayer&&window.dataLayer.push({event:\\\"gaEvent\\\",gaEventCategory:\\\"airpr\\\",gaEventAction:d+\\\"_pageview\\\",gaEventLabel:e,gaEventValue:0,gaEventNonInteractionBool:!0})}}},{}],2:[function(d,e,t){\\\"use strict\\\";var n=$(\\\".button-continue\\\"),i=$(\\\".auto-refund-input\\\"),o=$(\\\".auto-refund-form\\\");function a(){var t=!0;return i.each(function(){var d,e=$(this);\\\"\\\"===e.val()?(t&&(d=e)&&$(\\\"html,body\\\").animate({scrollTop:d.offset().top-36},333),t=!1,(d=e).addClass(\\\"error\\\"),d.prev(\\\"label\\\").addClass(\\\"error\\\")):((e=e).removeClass(\\\"error\\\"),e.prev(\\\"label\\\").removeClass(\\\"error\\\"))}),t}e.exports=function(){n.click(function(d){d.preventDefault(),a()&&o.submit()})}},{}],3:[function(d,e,t){\\\"use strict\\\";t.initDrawers=function(){var o=\\\"drawer-open\\\",a=\\\"drawer-transition\\\";$(document).on(\\\"click\\\",\\\"[data-drawer]\\\",function(){var d,e,t,n,i=$(this).attr(\\\"aria-controls\\\");return i=(e=(d=i).trim().split(/\\\\s+/)).map(function(d){return\\\"#\\\"+d}).join(\\\", \\\"),e=e.map(function(d){return'[data-drawer-indicator=\\\"'+d+'\\\"]'}).join(\\\", \\\"),i=$(i),t=$('[aria-controls=\\\"'+d+'\\\"]'),n=$(e).add(t).add(i),e=\\\"true\\\"===$(t[0]).attr(\\\"aria-expanded\\\"),n.addClass(a),e?i.slideUp(500,function(){t.attr(\\\"aria-expanded\\\",\\\"false\\\"),n.removeClass(a),n.removeClass(o)}):(i.slideDown(500,function(){t.attr(\\\"aria-expanded\\\",\\\"true\\\")}),n.addClass(o)),!1})},t.fdaModalSetUp=function(){var d=\\\"fda-main\\\",e=$(window),t=$(\\\".fda-info\\\"),n=$(\\\".fda-tabs\\\"),i=$(\\\".checkout-header\\\");0<d.length?e.scroll(_.debounce(function(){0<$(this).scrollTop()?(n.addClass(\\\"modal-scrolled\\\"),i.addClass(\\\"desktop-scrolled\\\")):(n.removeClass(\\\"modal-scrolled\\\"),i.removeClass(\\\"desktop-scrolled\\\"))},10)):t.scroll(_.debounce(function(){0<$(this).scrollTop()?(t.addClass(\\\"modal-scrolled-content\\\"),n.addClass(\\\"modal-scrolled\\\")):(t.removeClass(\\\"modal-scrolled-content\\\"),n.removeClass(\\\"modal-scrolled\\\"))},10));var o=$(\\\".fda-main\\\");o.removeClass(\\\"hide\\\"),$(\\\".fda-main-options-option\\\").click(function(d){d.preventDefault();var e,d=$(this).data(\\\"fda-option\\\");(e=$(\\\".fda-tabs [aria-controls=\\\"+d+\\\"]\\\")).trigger(\\\"click\\\"),o.fadeOut(333,function(){t.fadeIn(333),e.focus()})}),$(\\\".fda-tabs-tab\\\").click(function(){$(\\\".carrier-status-fda-container\\\").animate({scrollTop:0},250,\\\"swing\\\")})},t.initTabs=function(){var n=\\\"is-current-tab\\\",i=\\\"current-content\\\",o=37,a=38,r=39,A=40;function N(d){return d.closest('[role=\\\"tablist\\\"]')}function t(d){d=N(d);return $('[role=\\\"tab\\\"]',d)}function d(d){d.preventDefault();var d=$(this),e=d.closest(\\\".tabs-group\\\").attr(\\\"tab-group-tabs\\\");$(\\\"[tab-group-tab=\\\"+e+\\\"]\\\").not(d).removeClass(n).attr({\\\"aria-selected\\\":\\\"false\\\",tabindex:\\\"-1\\\"}),d.addClass(n),d.attr({\\\"aria-selected\\\":\\\"true\\\",tabindex:\\\"0\\\"}),d.focus();var d=d[0].hasAttribute(\\\"tabs-dropdown\\\")?d.val():d.attr(\\\"aria-controls\\\"),t=\\\"[tab-content=\\\"+e+\\\"-\\\"+d+\\\"]\\\";$(\\\"[tab-group-content=\\\"+e+\\\"]\\\").removeClass(i),setTimeout(function(){$(\\\"[tab-group-content=\\\"+e+\\\"]\\\").addClass(\\\"hide\\\"),$(t).removeClass(\\\"hide\\\"),setTimeout(function(){$(t).addClass(i)},33)},300)}function s(d){var e=t(d),d=e.index(d);(d+1<e.length?e.eq(d+1):e.eq(0)).focus()}function l(d){var e=t(d),d=e.index(d);(0<=d-1?e.eq(d-1):e.eq(e.length-1)).focus()}$(\\\"body\\\").unbind(\\\"click\\\").on(\\\"click\\\",\\\"[tabs-tab]\\\",d),$(\\\"body\\\").on(\\\"change\\\",\\\"[tabs-dropdown]\\\",d),$(\\\"body\\\").on(\\\"keydown\\\",\\\"[tabs-tab]\\\",function(d){var e=$(this),t=d.which;\\\"vertical\\\"===N(e).attr(\\\"aria-orientation\\\")?t===A?(d.preventDefault(),s(e)):t===a&&(d.preventDefault(),l(e)):t===r?s(e):t===o&&l(e)}),$(\\\"[tab-group-tabs]\\\").each(function(){var d,e;$(this)[0].hasAttribute(\\\"tab-default-id\\\")&&(d=$(this).attr(\\\"tab-default-id\\\"),e=$(this).attr(\\\"tab-group-tabs\\\"),$(\\\"[tab-group-tabs=\\\"+e+\\\"] [aria-controls=\\\"+d+\\\"]\\\").trigger(\\\"click\\\"))}),$(\\\"#brca, #arsacs\\\").addClass(\\\"current-content\\\").removeClass(\\\"hide\\\")}},{}],4:[function(d,e,t){\\\"use strict\\\";var i=d(\\\"./shlex\\\").parse_discount_info;const r=$(\\\".checkout-panel\\\");let A=null,o=null;var a=$(\\\".submit\\\"),N=$(\\\".js-expand-checkout-button\\\"),s=$(\\\".js-paypal-button\\\"),l=null;function c(){var d=TTAM.LANGUAGE_CODE.split(\\\"-\\\");return d[1]=d[1].toUpperCase(),d.join(\\\"-\\\")}function u(){var d=$(\\\".cartv2-items\\\");return $(\\\".js-add-product\\\").off(),$(\\\".js-remove-product\\\").off(),$(\\\".js-upgrade-product\\\").off(),d.empty(),d}function n(d){$(\\\".js-show-overlay\\\").off(),$(\\\".js-hide-overlay\\\").off(),$(\\\".learn-more-overlay\\\").off(),$(\\\".learn-more-overlay-wrapper\\\").off();var n=u();0<d.kits.length?($.each(d.kits,function(d,e){e.discount_info=i(e.discount_label);var t=$(\\\"#cartv2-\\\"+e.alias+\\\"-template\\\").html(),e=$(_.template(t,e));n.append(e)}),$(\\\"#cart-section-panel-left-container\\\").removeClass(\\\"empty\\\"),d.totals.items_discount?($(\\\".js-totals-subtotal\\\").removeClass(\\\"hide-discount-info\\\"),$(\\\".js-totals-savings\\\").removeClass(\\\"hide-discount-info\\\")):($(\\\".js-totals-subtotal\\\").addClass(\\\"hide-discount-info\\\"),$(\\\".js-totals-savings\\\").addClass(\\\"hide-discount-info\\\")),l()):($(\\\"#cart-section-panel-left-container\\\").addClass(\\\"empty\\\"),$(\\\".js-totals-subtotal\\\").addClass(\\\"hide-discount-info\\\"),$(\\\".js-totals-savings\\\").addClass(\\\"hide-discount-info\\\")),$(\\\".js-totals-subtotal-numkits\\\").text(\\\"Subtotal (\\\"+d.kits.length+\\\")\\\"),$(\\\".js-totals-subtotal-amount\\\").text(d.totals.items_total).formatCurrency({region:c()}),$(\\\".js-totals-savings-amount\\\").text(d.totals.items_discount).formatCurrency({region:c()}),$(\\\".js-totals-total-numkits\\\").text(\\\"Total (\\\"+d.kits.length+\\\")\\\"),$(\\\".js-totals-total-amount\\\").text(d.totals.items_total_with_discount).formatCurrency({region:c()}),a.prop(\\\"disabled\\\",0==d.kits.length),N.prop(\\\"disabled\\\",0==d.kits.length),s.prop(\\\"disabled\\\",0==d.kits.length),p(o),g(),o.emit(\\\"loading:stop\\\")}function p(n){$(\\\".js-add-product\\\").on(\\\"click\\\",function(d){n.emit(\\\"loading\\\");const e=$(d.currentTarget);var t=e.data(\\\"alias\\\").trim(),d=e.data(\\\"kit-url\\\");return $.ajax({type:\\\"POST\\\",url:d,data:{alias:t},dataType:\\\"json\\\"}).done(function(d){n.emit(\\\"cartv2:kit:operation\\\",d)}),!1}),$(\\\".js-remove-product\\\").on(\\\"click\\\",function(d){n.emit(\\\"loading\\\");const e=$(d.currentTarget);d=e.data(\\\"kit-url\\\");return $.ajax({type:\\\"DELETE\\\",url:d,dataType:\\\"json\\\"}).done(function(d){n.emit(\\\"cartv2:kit:operation\\\",d)}),!1}),$(\\\".js-upgrade-product\\\").on(\\\"click\\\",function(d){n.emit(\\\"loading\\\");const e=$(d.currentTarget);d=e.data(\\\"kit-url\\\");return $.ajax({type:\\\"POST\\\",url:d,dataType:\\\"json\\\"}).done(function(d){n.emit(\\\"cartv2:kit:operation\\\",d)}),!1})}const f=function(){$(\\\".expanded-checkout-wrapper .horizontal-button, #checkout-panel-mobile-veil\\\").off(\\\"click\\\"),$(\\\".expand-checkout-wrapper\\\").removeClass(\\\"expanded\\\"),$(\\\".expanded-checkout-wrapper\\\").removeClass(\\\"expanded\\\"),$(\\\"#checkout-panel-mobile-veil\\\").hide()};function h(){const e=r.offset().top;let t=$(window).width()<768;function d(){var d=t;t=$(window).width()<768,$(window).width()<768?(t&&!d&&$(window).unbind(\\\"scroll\\\"),r.addClass(\\\"sticky-enabled\\\").removeClass(\\\"sticky-disabled\\\").css(\\\"width\\\",\\\"100%\\\")):(f(),r.addClass(\\\"sticky-reset-for-calculation\\\"),r.css(\\\"width\\\",\\\"unset\\\"),d=r.outerWidth(),r.removeClass(\\\"sticky-reset-for-calculation\\\"),function(d){const t=function(){r.addClass(\\\"sticky-disabled\\\").removeClass(\\\"sticky-enabled\\\")},n=function(){r.addClass(\\\"sticky-enabled\\\").removeClass(\\\"sticky-disabled\\\")};n(),r.css(\\\"width\\\",d);const i=$(\\\"#cartv2-body\\\"),o=i.offset().top;var a=$(window).scrollTop();A=function(){var d=o+i.outerHeight(),e=r.offset().top+r.outerHeight();d<=e&&\\\"fixed\\\"===r.css(\\\"position\\\")?(a=$(window).scrollTop()-(e-d),t()):\\\"absolute\\\"===r.css(\\\"position\\\")&&$(window).scrollTop()<a&&n()},$(window).bind(\\\"scroll\\\",A)}((e,d)))}d(),$(window).resize(d)}function g(){function e(){return d.removeClass(\\\"is-open\\\"),!1}const d=$(\\\".learn-more-overlay\\\");$(\\\".js-show-overlay\\\").click(function(d){const e=$(d.currentTarget),t=$(\\\"#\\\"+e.data(\\\"overlay-id\\\"));return t&&t.addClass(\\\"is-open\\\"),!1}),$(\\\".js-hide-overlay\\\").click(e),$(document).on(\\\"keydown\\\",function(d){\\\"Escape\\\"==d.key&&e()}),$(\\\".learn-more-overlay\\\").click(function(){e()}),$(\\\".learn-more-overlay-wrapper\\\").click(function(d){d.stopPropagation()})}t.initialize=function(d,e){TTAM.cartv2&&(o=d,g(),p(d),d.on(\\\"cartv2:kit:operation\\\",n),$(\\\".product-cards.slick-carousel\\\").slick({arrows:!1,infinite:!1,centerMode:!0,variableWidth:!0,initialSlide:1,mobileFirst:!0,edgeFriction:0}),h(),$(\\\".js-expand-checkout-button, .expand-checkout-wrapper .horizontal-button\\\").on(\\\"click\\\",function(){$(\\\".expand-checkout-wrapper\\\").addClass(\\\"expanded\\\"),$(\\x�ssl_sniffer: we might lose some data (%d), need some recursive read
Segmentation fault
```

--- a/src/ssl_sniffer/ebpf/main.bpf.c
+++ b/src/ssl_sniffer/ebpf/main.bpf.c
@@ -66,7 +66,7 @@ static __always_inline int handle_rw_exit(struct pt_regs *ctx, int is_write)
     bpf_get_current_comm(&msg->comm, TASK_COMM_LEN);
     bpf_core_read(&msg->ts, sizeof(msg->ts), &ts);
     msg->op = is_write ? SSL_OP_WRITE : SSL_OP_READ;
-    msg->len = resp;
+    msg->len = read_len;
 
     //if (!is_write)
     //    bpf_probe_write_user((void *)*buf, "HTTP/1.1 200 OK\nContent-Length: 12\n\nHello World\n\00", 50);

